Python/Product/Analyzer/Intellisense/OutOfProcProjectAnalyzer.cs

```diff
@@ -1,2381 +1,2139 @@
-            var entry = _projectFiles.Get<IPythonProjectEntry>(completions.fileId);
-            if (entry == null) {
-                return IncorrectFileType();
-            }
-
-            IEnumerable<MemberResult> members;
-            if (entry.Analysis != null) {
-                members = entry.Analysis.GetMembers(
-                    completions.text,
-                    new SourceLocation(completions.line, completions.column),
-                    completions.options
-                ).MaybeEnumerate();
-            } else {
-                members = Enumerable.Empty<MemberResult>();
-            }
-
-            return new AP.CompletionsResponse() {
-                completions = ToCompletions(members.ToArray(), completions.options)
-            };
-        }
-
-        private static IEnumerable<MemberResult> GetModuleVariables(
-            IPythonProjectEntry entry,
-            GetMemberOptions opts,
-            string prefix
-        ) {
-            var analysis = entry?.Analysis;
-            if (analysis == null) {
-                yield break;
-            }
-
-            foreach (var m in analysis.GetAllAvailableMembers(SourceLocation.None, opts)) {
-                if (m.Values.Any(v => v.DeclaringModule == entry)) {
-                    if (string.IsNullOrEmpty(prefix) || m.Name.StartsWith(prefix, StringComparison.OrdinalIgnoreCase)) {
-                        yield return m;
-                    }
-                }
-            }
-        }
-
-        private Response GetAllMembers(Request request) {
-            var req = (AP.GetAllMembersRequest)request;
-
-            var members = Enumerable.Empty<MemberResult>();
-            var opts = GetMemberOptions.ExcludeBuiltins | GetMemberOptions.DeclaredOnly | req.options;
-
-            foreach (var entry in _projectFiles) {
-                members = members.Concat(
-                    GetModuleVariables(entry.Value as IPythonProjectEntry, opts, req.prefix)
-                );
-            }
-
-            members = members.GroupBy(mr => mr.Name).Select(g => g.First());
-
-            return new AP.CompletionsResponse() {
-                completions = ToCompletions(members.ToArray(), opts)
-            };
-        }
-
-        private AP.Signature[] ToSignatures(IEnumerable<IOverloadResult> sigs) {
-            return sigs.Select(
-                sig => new AP.Signature() {
-                    name = sig.Name,
-                    doc = sig.Documentation,
-                    parameters = sig.Parameters.Select(
-                        param => new AP.Parameter() {
-                            name = param.Name,
-                            defaultValue = param.DefaultValue,
-                            optional = param.IsOptional,
-                            doc = param.Documentation,
-                            type = param.Type,
-                            variables = param.Variables?.Select(MakeReference).Where(r => r != null).ToArray()
-                  
```

Comment:
`oldExtension` can be disposed outside of the critical section. Not sure how critical is it.